clang/include/clang/ASTMatchers/ASTMatchers.h

```diff
@@ -3076,6 +3076,21 @@ inline internal::BindableMatcher<Stmt> sizeOfExpr(
 /// \code
 ///   namespace a { namespace b { class X; } }
 /// \endcode
+///
+/// Qualified names in templated classes can be matched explicitly or implicity
+/// by specifying the template type or using `<*>` to match any template.
```

Comment:
We should probably be clear that `<*>` is the *only* syntax we support. e.g., we don't let you search based on arity, as in `hasName("Foo<int, *>::Bar")`. Might be worth an example as well, definitely worth some test coverage.